### DISCUSSION TITLE
rust analyzer not working

### DISCUSSION THREAD

**Discussion by sfgxdfdfg** [zed-industries/zed#15719] [Feature Requests]
Hello. I installed zed on my nixos with declaring it in my configuration.nix and when i open it and open a rust project the rust analyzer not  showing up and the code completion not working and it always says click to restart and update  im clicking but nothing happens.

**Comment by JuliusDeBoer** [comment]
I believe this is a known issue with Zed on NixOS (https://wiki.nixos.org/wiki/Zed).

> *"By default, Zed will try to download pre-built LSP servers in* `~/.local/share/zed/languages/`*. This does not work for NixOS."*

Im guessing a possible solution is to get the LSP from `$PATH`, which the NixOS wiki mentions. Or maybe provide an option to set the directory where LSP's get downloaded to.

**Comment by k3yss** [comment]
+ 1 for this

**Comment by oliveirarleo** [comment]
Simply compiling to musl could solve this. I saw that the new ssh server works like that, why wont the binaries shipped with zed be compiled for musl? other distros could profit from that too.
I don't see why not use all binaries with musl, but if not, then maybe an option in the settings json could make this work.

**Comment by pxp9** [comment]
hey you should try this snippet

```nix
lsp = {
    rust-analyzer = {

        binary = {
            #                        path = lib.getExe pkgs.rust-analyzer; # you can use this one if you prefer,
            path_lookup = true;
        };
    };
};
```

note I am using home-manager to configure zed and the option `userSettings` which basically translates nix to `json` and put the conf read-only.

**Reply by pxp9** [reply]
take a look how it is done in simple json

https://github.com/zed-industries/zed/pull/12418#issuecomment-2137245977

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
